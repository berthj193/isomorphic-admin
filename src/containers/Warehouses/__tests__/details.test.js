import React from 'react';
import WarehouseDetails, { WarehouseDetails as RawWarehouseDetails } from '../details';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<WarehouseDetails />', () => {
  const defaultParams = {
    match: {
      params: {},
    },
    warehouses: [{
      id: 1,
      name: 'foo',
      export_class: 'foo',
      stores: [{}],
      shipping_rates: [{}],
      options: [{}],
    }],
  };
  const getComponent = (props = {}) => shallow(<RawWarehouseDetails {...defaultParams} {...props} />);
  const mockStore = configureStore();

  it('should render properly', () => {
    expect(toJson(getComponent({ orders: [{ id: 1 }] }))).toMatchSnapshot();
  });

  it('should render loading page', () => {
    expect(toJson(getComponent({ warehouses: [{}] }))).toMatchSnapshot();
  });

  it('renderList should return list of elements', () => {
    expect(toJson(getComponent().instance().renderList(['foo', 'bar'])())).toMatchSnapshot();
  });

  it('renderOptions should return list of key value options', () => {
    expect(toJson(getComponent().instance().renderOptions([{ key: 'foo', value: 'bar' }])())).toMatchSnapshot();
  });

  it('loadDetailList should return list of details to render', () => {
    expect(getComponent().instance().loadDetailList().length).toBeGreaterThan(1);
  });

  it('should render details with provider', () => {
    expect(toJson(shallow(<WarehouseDetails store={mockStore()} />))).toMatchSnapshot();
  });
});
