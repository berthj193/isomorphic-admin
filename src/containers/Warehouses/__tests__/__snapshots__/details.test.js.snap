// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WarehouseDetails /> renderList should return list of elements 1`] = `undefined`;

exports[`<WarehouseDetails /> renderOptions should return list of key value options 1`] = `undefined`;

exports[`<WarehouseDetails /> should render details with provider 1`] = `
<ComponentWithProvider
  startRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`<WarehouseDetails /> should render loading page 1`] = `
<React.Fragment>
  <LayoutWrapper
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <LayoutContent>
      <h1>
        Warehouses
      </h1>
      <ButtonList
        buttons={
          Array [
            Object {
              "link": "/dashboard/warehouses",
              "title": "Go back",
            },
            Object {
              "link": "/dashboard/warehouses/undefined/edit",
              "title": "Edit",
            },
          ]
        }
      />
      <PageDetails
        details={
          Array [
            Object {
              "Name": undefined,
            },
            Object {
              "Export class": undefined,
            },
            "---",
            false,
            false,
            false,
            false,
            false,
            false,
            Object {
              "Updated at": undefined,
            },
          ]
        }
      />
    </LayoutContent>
  </LayoutWrapper>
</React.Fragment>
`;

exports[`<WarehouseDetails /> should render properly 1`] = `
<React.Fragment>
  <LayoutWrapper
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <LayoutContent>
      <h1>
        Warehouses
      </h1>
      <ButtonList
        buttons={
          Array [
            Object {
              "link": "/dashboard/warehouses",
              "title": "Go back",
            },
            Object {
              "link": "/dashboard/warehouses/undefined/edit",
              "title": "Edit",
            },
          ]
        }
      />
      <PageDetails
        details={
          Array [
            Object {
              "Name": "foo",
            },
            Object {
              "Export class": "foo",
            },
            "---",
            Object {
              "Options": [Function],
            },
            "---",
            Object {
              "Shipping rates": [Function],
            },
            "---",
            Object {
              "Stores": [Function],
            },
            "---",
            Object {
              "Updated at": undefined,
            },
          ]
        }
      />
    </LayoutContent>
  </LayoutWrapper>
</React.Fragment>
`;
